controller/engine_controller.go

```diff
@@ -1948,6 +1948,44 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replicaName, add
 				"Start rebuilding replica %v with Address %v for normal engine %v and volume %v", replicaName, addr, e.Name, e.Spec.VolumeName)
 			err = engineClientProxy.ReplicaAdd(e, replicaName, replicaURL, false, fastReplicaRebuild, localSync, fileSyncHTTPClientTimeout, grpcTimeoutSeconds)
 		}
+
+		// For v2 engine, the rebuilding is an async call. We need to wait for the rebuilding start then complete here
```

Comment:
Is the resource reconciliation stuck here until the rebuild is complete? If the controller has 5 workers and all are waiting for v2 rebuilding, does this mean no workers are available to reconcile engine resources? Should we use a goroutine for the asynchronous call?